Userland/Libraries/LibGL/SoftwareGLContext.cpp

```diff
@@ -686,9 +680,13 @@ void SoftwareGLContext::gl_delete_textures(GLsizei n, const GLuint* textures)
     for (auto i = 0; i < n; i++) {
         GLuint name = textures[i];
 
-        // unbind texture if it is currently bound
-        if (m_bound_texture_2d == name)
-            m_bound_texture_2d = 0;
+        // Check all texture units
+        for (auto& texture_unit : m_texture_units) {
+            auto texture_object = m_allocated_textures.find(name);
+
+            if (texture_object->value == texture_unit.bound_texture())
+                texture_unit.unbind_texture(GL_TEXTURE_2D);
+        }
```

Comment:
```suggestion
        auto texture_object = m_allocated_textures.find(name);
        if (texture_object == m_allocated_texture.end() || texture_object->value == nullptr)
            continue;

        for (auto& texture_unit : m_texture_units) {
            if (texture_object->value == texture_unit.bound_texture())
                texture_unit.unbind_texture(GL_TEXTURE_2D);
        }
```
The name might not actually exist, because it was never allocated via `glGenTextures` also if the name was never bound to a unit before its `value` will be `nullptr` and we can skip iterating all units.